corpus/codeblock.txt

```diff
@@ -89,13 +93,51 @@ text
       (word))
     (line
       (codeblock
-        (line)
-        (line)
-        (line))))
+        (code
+          (line)
+          (line)
+          (line)))))
   (block
     (line
       (word))))
 
+================================================================================
+codeblock with language annotations
+================================================================================
+This code is in Lua: >lua
+  local foo = 'bar'
+<
+
+This code is in Vimscript: >vim
+  let foo = "bar"
+<
+
+--------------------------------------------------------------------------------
+
+(help_file
+  (block
+    (line
+      (word)
+      (word)
+      (word)
+      (word)
+      (word)
+      (codeblock
+        (language)
+        (code
```

Comment:
- why is a new child node needed?
- would it make sense for `language` to be a "field" instead of a node?

Reply:
It's definitely an alternative option I've played with; I don't know why that would be better. The parser seems to have slightly fewer states this way, which swayed me (less complexity).

Reply:
And what would be the token you would put in the `field`? `word` doesn't work, due to conflicts and the included whitespace.

Reply:
I thought fields were for labeling metadata on nodes; the "language" of a codeblock is a property of the codeblock, not a child of the codeblock. 

But reading https://tree-sitter.github.io/tree-sitter/creating-parsers#using-fields makes me doubt this.

Reply:
Yes. The point is that we need a _new_ type of node and cannot reuse an existing node (like `word`), which is what a `field` would be for. (Unless I'm missing some treesitter magic here, which is very possible.)